src/main.rs

```diff
@@ -325,6 +374,12 @@ fn cargo_careful(mut args: env::Args) {
         "CARGO_ENCODED_RUSTDOCFLAGS",
         rustc_build_sysroot::encode_rustflags(&flags),
     );
+
+    // Leaks are not a memory safety issue, don't detect them by default
+    if sanitizer == Some("address") && !env::vars_os().any(|(key, _)| key == "ASAN_OPTIONS") {
```

Comment:
Why iterate over all env vars? `env::var_os("ASAN_OPTIONS).is_some()` makes a lot more sense, doesn't it?